src/Lean/Meta/Injective.lean

```diff
@@ -135,14 +135,15 @@ def mkInjectiveTheoremNameFor (ctorName : Name) : Name :=
 private def mkInjectiveTheorem (ctorVal : ConstructorVal) : MetaM Unit := do
   let some type ← mkInjectiveTheoremType? ctorVal
     | return ()
-  let value ← mkInjectiveTheoremValue ctorVal.name type
   let name := mkInjectiveTheoremNameFor ctorVal.name
-  addDecl <| Declaration.thmDecl {
-    name
-    levelParams := ctorVal.levelParams
-    type        := (← instantiateMVars type)
-    value       := (← instantiateMVars value)
-  }
+  realizeConst ctorVal.name name do
```

Comment:
Should we attach them to the name of the first inductive of the group, so that all realizable constructions about this inductive can realize other constructions, whether they are per constructor or wholesome? (I hope I remember the mechanism correctly.)